### PR TITLE
Documentation spelling errors

### PR DESCRIPTION
**Pull request by doreis-lowrisc**
Fixes tinies spelling erros in the documentation.

### CODE REVIEW

File: hw/ip/kmac/doc/_index.md [hunk 1]
```diff
@@ -53,7 +53,7 @@ Based on the given criteria, the MSG_FIFO depth was determined to store the inco
 
 The MSG_FIFO has a packer in front.
 It packs any partial writes into the size of inernal datapath (64bit) and stores in MSG_FIFO.
-It gives some freedome to the software not requiring to align the messages.
+It gives some freedom to the software not requiring to align the messages.
```
Comment (luismarques) [inline hw/ip/kmac/doc/_index.md:56]: As long as you're improving this, feel free to also tweak the grammar. Maybe "It frees the software from having to align the messages", or something like that.

File: hw/ip/kmac/doc/_index.md [hunk 2]
```diff
@@ -53,7 +53,7 @@ Based on the given criteria, the MSG_FIFO depth was determined to store the inco
 
 The MSG_FIFO has a packer in front.
 It packs any partial writes into the size of inernal datapath (64bit) and stores in MSG_FIFO.
```
Comment (luismarques) [inline hw/ip/kmac/doc/_index.md:55]: Also: inernal, internal?

Comment (doreis-lowrisc) [reply hw/ip/kmac/doc/_index.md:55]: good catch!

File: hw/ip/kmac/doc/_index.md (hunk 1 above)
Comment (doreis-lowrisc) [reply hw/ip/kmac/doc/_index.md:56]: much better!

**Comment by doreis-lowrisc** [comment]
@luismarques do you have any adea of the meaning of Comportable devices?
https://github.com/lowRISC/opentitan/pull/9768/files/335e8a3ed6ee89e22047d321d2ad630ccd2e1f54#diff-b42a35a5e243a3461808996cd9fafba6c7b8eda52ae46705bcf005f29beaa701L29

**Comment by luismarques** [comment]
> @luismarques do you have any adea of the meaning of Comportable devices? https://raw.githubusercontent.com/lowRISC/opentitan/master/hw/ip/tlul/doc/_index.md#L29

See <https://docs.opentitan.org/doc/rm/comportability_specification/>. It's basically our standardized component approach.

**Review by rswarbrick — APPROVED** [review]

**Pull request merged**